gframe/data_manager.cpp

```diff
@@ -343,6 +344,16 @@ const CardDataC* DataManager::GetMappedCardData(uint32_t code) const {
 		return gDataManager->GetCardData(it->second);
 	return nullptr;
 }
+std::string DataManager::GetNameString(uint32_t code) const {
+	std::wstring name;
+	auto csit = cards.find(code);
+	if(csit == cards.end() || csit->second.GetStrings().name.empty())
+		name = {};
+	name = csit->second.GetStrings().name;
+
+	std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t> converter;
+	return converter.to_bytes(name);
```

Comment:
This function should return a wstringview like the others, without allocating temp strings, and for the conversion between utf16/32 and utf8, use the `BufferIO` family of functions, in this case you'd want to use `BufferIO::EncodeUTF8` (but in the caller)